scrnaseq.Rmd

```diff
@@ -1266,60 +1378,61 @@ Take care not to mis-read a UMAP:
 For a nice read to intuitively understand UMAP, see `r Cite("https://pair-code.github.io/understanding-umap/")`. 
 
 ## Coloured by cluster
-```{r umap_by_cluster}
+
+```{r run_umap}
 # Default UMAP
 sc = suppressWarnings(Seurat::RunUMAP(sc, dims=1:param$pc_n, verbose=FALSE, umap.method="uwot"))
+```
 
-# 3D UMAP
-sc = suppressWarnings(Seurat::RunUMAP(sc, dims=1:param$pc_n, n.components=3, reduction.name="umap3d", reduction.key="UMAP3D_", verbose=FALSE, umap.method="uwot"))
-
+```{r umap_by_cluster, fig.asp=0.6}
 # Note that you can set `label = TRUE` or use the LabelClusters function to help label individual clusters
 cluster_cells = table(sc@active.ident)
 cluster_labels = paste0(levels(sc@active.ident)," (", cluster_cells[levels(sc@active.ident)],")")
-p = Seurat::DimPlot(sc, reduction="umap", label=TRUE) + 
+p = Seurat::DimPlot(sc, reduction="umap", label=TRUE, label.box=TRUE) + 
   AddStyle(title="UMAP, cells coloured by cluster identity", legend_position="bottom", legend_title="Clusters") + 
-  scale_color_manual(values=param$col_clusters, labels=cluster_labels)
+  scale_color_manual(values=param$col_clusters, labels=cluster_labels) +
+  scale_fill_manual(values=param$col_clusters, labels=cluster_labels) +
+  theme(legend.position="none")
 p
 ```
 
 ## Coloured by sample 
-```{r umap_by_sample}
+```{r umap_by_sample, fig.asp=0.6}
 # Add a UMAP that is coloured by sample of origin
 cell_samples = sc[[]] %>% dplyr::pull(orig.ident) %>% unique() %>% sort()
 
 # Note: This is a hack to colour by sample but label by Cluster
-p = Seurat::DimPlot(sc, group.by="orig.ident", pt.size=1, cols=param$col_samples) +
+p = Seurat::DimPlot(sc, reduction="umap", group.by="orig.ident", cols=param$col_samples) +
   AddStyle(title="UMAP, cells coloured by sample of origin", legend_position="bottom")
 p$data$seurat_clusters = sc[["seurat_clusters"]][rownames(p$data), ]
-p = LabelClusters(p, id="seurat_clusters")
+p = LabelClusters(p, id="seurat_clusters", box=TRUE, segment.color="black", fill="lightgrey")
```

Comment:
Can we keep the boxes for the large plots, and skip 'em for the small plots?

Reply:
ok